### PR TITLE
[MA-38] Added filter/validation queries for wpcids of active keywords

### PR DESCRIPTION
**Pull request by jellyvora-reachlocal**
Build: https://jenkins.media.dev.usa.reachlocalservices.com/view/media-automation/job/media-automation-build/68/

### CODE REVIEW

File: src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties [hunk 1]
```diff
@@ -0,0 +1,49 @@
+GET_ACTIVE_KEYWORDS_WEB_PUBLISHER_CAMPAIGN_IDS = \
+SELECT WPCHK.WebPublisherCampaign_idWebPublisherCampaign_FK, WPCHK.KeywordMatchType_idKeywordMatchType_FK, \
+  WPCHK.AdGroup_idAdGroup_FK, SC.WebPublisher_idWebPublisher_FK \
+FROM WebPublisherCampaign WPC \
+JOIN SubCampaign SC ON SC.idSubCampaign = WPC.SubCampaign_idSubCampaign_FK \
+JOIN WebPublisherCampaign_has_Keyword WPCHK \
+  ON WPCHK.WebPublisherCampaign_idWebPublisherCampaign_FK = WPC.idWebPublisherCampaign \
+JOIN Offer o \
+  ON o.idOffer = WPC.Offer_idOffer_FK \
+WHERE SC.Campaign_idCampaign_FK = :campaignId \
```
Comment (balkrishnada) [inline src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties]: Doing more research with the tables, it looks we should include Match Type as well in the condition to be sure enough. Let me confirm with Jonathan what does a NULL match type indicate in the BI table. I think they should match to 'Broad`.

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        @Value("${GET_WEB_PUBLISHER_CAMPAIGN_IDS_IN_KEYWORDS_UPDATE}") String wpcIdsHavingKeywordInUpdateQuery,
+        @Value("${GET_WEB_PUBLISHER_CAMPAIGN_IDS_IN_GEO_KEYWORDS_UPDATE}") String wpcIdsHavingGeoKeywordInUpdateQuery) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.wpcIdsHavingActiveKeywordQuery = wpcIdsHavingActiveKeywordQuery;
+        this.wpcIdsHavingActiveGeoKeywordQuery = wpcIdsHavingActiveGeoKeywordQuery;
+        this.wpcIdsHavingKeywordInUpdateQuery = wpcIdsHavingKeywordInUpdateQuery;
+        this.wpcIdsHavingGeoKeywordInUpdateQuery = wpcIdsHavingGeoKeywordInUpdateQuery;
+    }
+
+    public List<WebPublisherCampaignKeyword> getActiveKeywordWebPublisherCampaignIds(Long campaignId, Long keywordId) {
```
Comment (balkrishnada) [inline src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: `getWebPublisherCampaignIdsForActiveKeywords`

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        Map<String, Object> params = new HashMap<>();
+        params.put(CID_PARAM, campaignId);
+        params.put(KWD_ID_PARAM, keywordId);
+        return jdbcTemplate.query(wpcIdsHavingActiveKeywordQuery, params, this::webPublisherCampaignKeywordRowMapper);
+    }
+
+    public List<WebPublisherCampaignKeyword> getActiveGeoKeywordWebPublisherCampaignIds(Long campaignId,
```
Comment (balkrishnada) [inline src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: `getWebPublisherCampaignIdsForActiveGeoKeywords`

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/auditlogs/AuditLog.java [hunk 4]
```diff
@@ -1,4 +1,6 @@
-package com.reachlocal.media.automation.jobs.removekeywords.steps.auditlogs;
+package com.reachlocal.media.automation.jobs.removekeywords.auditlogs;
+
+import com.reachlocal.media.automation.jobs.removekeywords.keywordupdate.KeywordMatchType;
 
 public class AuditLog {
```
[the remark this answers is not in the document]
Comment (jellyvora-reachlocal) [reply src/main/java/com/reachlocal/media/automation/jobs/removekeywords/auditlogs/AuditLog.java]: we need to have a column in low_qs_keyword_deletion_audit_log for `AdGroupId`

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                                                                                        Long geoKeywordId) {
+        Map<String, Object> params = new HashMap<>();
+        params.put(CID_PARAM, campaignId);
+        params.put(KWD_ID_PARAM, keywordId);
+        params.put(GEO_KWD_ID_PARAM, geoKeywordId);
+        return jdbcTemplate.query(wpcIdsHavingActiveGeoKeywordQuery, params,
+                                  this::webPublisherCampaignKeywordRowMapper);
+    }
+
+    public List<Long> getWebPublisherCampaignIdsInKeywordUpdate(List<Long> wpcIds, Long keywordId) {
```
Comment (balkrishnada) [inline src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: `getDeactivationKeywordsForWebPublisherCampaignIds`

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        Map<String, Object> params = new HashMap<>();
+        params.put(WPC_IDS_PARAM, wpcIds);
+        params.put(KWD_ID_PARAM, keywordId);
+        return jdbcTemplate.queryForList(wpcIdsHavingKeywordInUpdateQuery, params, Long.class);
+    }
+
+    public List<Long> getWebPublisherCampaignIdsInGeoKeywordUpdate(List<Long> wpcIds,
```
Comment (balkrishnada) [inline src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: `getDeactivationGeoKeywordsForWebPublisherCampaignIds`

File: src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  AND WPCHGHK.Keyword_idKeyword_FK= :keywordId \
+  AND WPCHGHK.GeoKeyword_idGeoKeyword_FK = :geoKeywordId \
+  AND WPCHGHK.KeywordActivationType_idKeywordActivationType_FK=1 \
+  AND o.use_keywords = 1 \
+  AND WPC.wpc_start_date IS NOT NULL \
+  AND WPC.wpc_end_date IS NULL
+
+
+GET_WEB_PUBLISHER_CAMPAIGN_IDS_IN_KEYWORDS_UPDATE = \
```
[the remark this answers is not in the document]
Comment (jellyvora-reachlocal) [reply src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties]: > WPC_HAS_KEYWORD_UPDATE_VALIDATION

Do we need `VALIDATION`,  also it is a check against all wpcIds of a keyword

[the remark this answers is not in the document]
Comment (balkrishnada) [reply src/main/resources/sql/removekeywords/validKeywordsWebPublisherCampaignIds.properties]: Good point, please feel free to tweak in line with the suggestion.

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/auditlogs/AuditLog.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (jellyvora-reachlocal) [reply src/main/java/com/reachlocal/media/automation/jobs/removekeywords/auditlogs/AuditLog.java]: I will add the column in V1__init sql file

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java (hunk 5 above)
Comment (jellyvora-reachlocal) [reply src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: That is not what the method does, it gets wpc ids for which keyword deletion entry is made in update table.

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java (hunk 6 above)
Comment (jellyvora-reachlocal) [reply src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: same as above

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java (hunk 5 above)
Comment (balkrishnada) [reply src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: That's exactly the name `getDeactivationKeywordsForWebPublisherCampaignIds` suggests? There is a hardcoded activation type = 2 being used. What other name you have in mind?

**Comment by jellyvora-reachlocal** [comment]
> good so far. some minor update to query will be needed based on feedback I will share later.
> Apart from that the queries also look good.

Addressed all the comments so far and updated build

File: src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    public List<Long> getWebPublisherCampaignIdsForDeletedKeyword(List<Long> wpcIds, Long keywordId) {
+        Map<String, Object> params = new HashMap<>();
+        params.put(WPC_IDS_PARAM, wpcIds);
+        params.put(KWD_ID_PARAM, keywordId);
+        return jdbcTemplate.queryForList(wpcIdsForDeletedKeywordQuery, params, Long.class);
+    }
+
+    public List<Long> getWebPublisherCampaignIdsInForDeletedGeoKeyword(List<Long> wpcIds,
```
Comment (balkrishnada) [inline src/main/java/com/reachlocal/media/automation/jobs/removekeywords/webpublishercampaignkeyword/WebPublisherCampaignKeywordRepository.java]: please remove `In`

**Pull request merged**